### ISSUE TITLE
Change in behaviour of yielding an observable between 2.5.3 and 3.0.0

### ISSUE THREAD

**Issue by mjolka** [Reactive-Extensions/RxJS#950]
In 2.5.3, yielding an observable returns the last item in the sequence. For example, the following will print 2 to the console:

    Rx.spawn(function* () {
      console.log(yield Rx.Observable.range(1, 2));
    })();

In 3.0.0 and above, yielding an observable returns the first item in the sequence. For example, the following will print 1:

    Rx.Observable.spawn(function* () {
      console.log(yield Rx.Observable.range(1, 2));
    }).subscribe();

I couldn't find this mentioned in the [release notes](https://github.com/Reactive-Extensions/RxJS/releases/tag/v3.0.0), so am wondering if the change in behaviour was intentional.

The behaviour in 2.5.3 is in line with the behaviour of Rx.NET, where the following will print 2:

    Observable.FromAsync(async () =>
        Console.WriteLine(await Observable.Range(1, 2))
    ).Subscribe();

**Comment by xgrommx** [comment]
I think instead of https://github.com/Reactive-Extensions/RxJS/blob/master/src/core/linq/observable/spawn.js#L45 should be `g.add(value.last().subscribe(processGenerator, onError))`. I checked on my local environment and it working as expected. If @mattpodwysocki will approve this approach I'll prepare PR for fix it.

**Comment by xgrommx** [comment]
A new approach should solve this problem:
```js
      function next(ret) {
        if (ret.done) {
          o.onNext(ret.value);
          o.onCompleted();
          return;
        }
        var obs = toObservable.call(self, ret.value);
        var value = null;
        if (Observable.isObservable(obs)) {
          g.add(obs.subscribe(function(val) {
            value = val;
          }, onError, function() {
            processGenerator(value);
          }));
        } else {
          onError(new TypeError('type not supported'));
        }
      }
```
Also I have some changes which give ability use this code:
```js
Rx.Observable.wrap(function* (value) {
    return yield [1,2,3,4,
        Rx.Observable.just(1000).delay(1000),
        {x: Rx.Observable.just(4000).delay(1000), y: Rx.Observable.just(3000)}, [3, 4, 5, Rx.Observable.just(5000)]];
})(3000).subscribe(x => console.log(x));
```
Now we can use mixed array with flat array and observables. What do you think @mattpodwysocki ?

**Comment by paulpdaniels** [comment]
At @xgrommx what is the advantage of 

```javascript

g.add(obs.subscribe(function(val) {
  value = val;
}, onError, function() {
  processGenerator(value);
}));

```

over your first example

`g.add(value.last().subscribe(processGenerator, onError));`

?

**Comment by paulpdaniels** [comment]
Ah nvm I see the problem.

```javascript
if (ret.done) {
  o.onNext(ret.value); //This is undefined.
  o.onCompleted();
  return;
}
```

**Comment by mattpodwysocki** [comment]
Fixed the issue for #950 and #952
